services/metadata/src/metadata/services/templates.clj

```diff
@@ -16,24 +18,39 @@
       (ca/assert-found "metadata template" template-id)
       (remove-nil-values)))
 
-(defn view-template-csv
-  [template-id]
+(defn- format-template-csv
+  [template-data]
   (csv/csv-string [
-    (->> (view-template template-id)
+    (->> template-data
       :attributes
       (map :name)
       (cons "file name or path"))]))
 
-(defn view-template-guide
+(defn view-template-csv
   [template-id]
+  (format-template-csv (view-template template-id)))
+
+(defn- format-template-guide
+  [template-data]
   (csv/csv-string
-    (->> (view-template template-id)
+    (->> template-data
       :attributes
       (map (juxt :name :description :required :type #(if (:values %) (string/join ", " (map :value (:values %))) "")))
       (cons ["attribute name", "attribute description",
              "required (If you cannot provide,  enter 'not collected', 'not applicable' or 'missing'.)",
              "value type definition", "enum value options (you must enter one of these values)"]))))
 
+(defn view-template-guide
+  [template-id]
+  (format-template-guide (view-template template-id)))
```

Comment:
Stuff above here is all refactoring to separate the formatting of the CSV files from template data from the fetching of the data, for the existing/prior endpoints.